include/eve/detail/function/simd/common/load.hpp

```diff
@@ -164,105 +68,189 @@ namespace eve::detail
-  EVE_FORCEINLINE auto load_(EVE_SUPPORTS(cpu_), as<Wide> const&, Iterator b, Iterator e) noexcept
+  template<callable_options O, std::input_iterator It, typename Wide>
+  EVE_FORCEINLINE Wide load_(EVE_REQUIRES(cpu_), O const&, It src, [[maybe_unused]] It end, as<Wide> tgt) noexcept
   {
-    return load(ignore_none, safe, as<Wide>{}, b, e);
+    if constexpr (logical_simd_value<Wide> && !Wide::abi_type::is_wide_logical)
+    {
+      return to_logical(load(src, end, as<as_arithmetic_t<Wide>>{}));
+    }
+    else
+    {
+      return piecewise_load(src, tgt);
+    }
   }
 
-  template<std::input_iterator Iterator, typename Pack>
-  EVE_FORCEINLINE Pack
-  load_ ( EVE_SUPPORTS(cpu_), ignore_none_ const&, safe_type const&
-        , eve::as<Pack> const &tgt, Iterator b, [[maybe_unused]] Iterator e
-        ) noexcept
+  template<detail::data_source DS, typename T, typename N>
+  EVE_FORCEINLINE logical<wide<T, N>> load_impl(cpu_, DS src, as<logical<wide<T, N>>> tgt) noexcept
   {
-    return piecewise_load(tgt,b);
+    if constexpr (logical<wide<T, N>>::abi_type::is_wide_logical)
+    {
+      return bit_cast(load(ptr_cast<T const>(src), as<wide<T, N>>{}), tgt);
+    }
+    else
+    {
+      return to_logical(load(ptr_cast<T const>(src), as<wide<T, N>>{}));
+    }
   }
 
-  //================================================================================================
-  // Load from pointer - Emulation
-  //================================================================================================
-  template<typename T, typename N, data_source Ptr>
-  EVE_FORCEINLINE wide<T, N> load_( EVE_SUPPORTS(cpu_), ignore_none_ const&, safe_type const&
-                                  , eve::as<wide<T, N>> const &tgt, Ptr ptr
-                                  ) noexcept
-  requires simd_compatible_ptr<Ptr,wide<T, N>> && std::same_as<abi_t<T, N>, emulated_>
+  template<relative_conditional_expr C, detail::data_source DS, typename Wide>
+  EVE_FORCEINLINE Wide load_cx_(C const& cx, DS src, as<Wide> tgt) noexcept
   {
-    return piecewise_load(tgt, unalign(ptr));
-  }
+    using e_t = typename pointer_traits<Wide>::value_type;
+    using c_t = cardinal_t<Wide>;
 
-  //================================================================================================
-  // Aggregation
-  //================================================================================================
-  template<typename T, typename N, data_source Ptr>
-  EVE_FORCEINLINE wide<T, N> load_( EVE_SUPPORTS(cpu_), ignore_none_ const&, safe_type const&
-                                  , eve::as<wide<T, N>> const &tgt, Ptr ptr
-                                  ) noexcept
-  requires(simd_compatible_ptr<Ptr,wide<T, N>> && std::same_as<abi_t<T, N>, aggregated_>)
-  {
-    return aggregate_load(tgt,ptr);
+    if constexpr (!std::is_pointer_v<DS>)
+    {
+      constexpr bool is_aligned_enough = c_t() * sizeof(e_t) >= DS::alignment();
```

Comment:
I'm so confused to what;'s happening.

DS is data_source. We use that in the API of load. Here we call ::alignmetn on it. We obv can't do that for pointers. What's happening.